### DISCUSSION TITLE
Apply "AND" of multiple linked PRs for a single issue to be closed or allow non-closing linked PRs

### DISCUSSION THREAD

**Discussion by achimnol** [community/community#9926] [Projects and Issues]
Currently the linked PRs can be connected only by stating that "it resolves/fixes/closes" the issue.
As #4619 suggests, I'd like to have an option to just "refer" an issue without closing it when merged.

Also, I'd like to set the closing condition to be the "AND" combination of all linked PRs when there are multiple linked PRs to a single issue. I have this pattern very frequently because the work scope of a single issue often spans across multiple different components (repositories) to fix up. Currently, closing just one of linked PRs closes the issue, when still there are open linked PRs, meaning that it is working as an "OR" combination.

Here is an example case:
![image](https://user-images.githubusercontent.com/555156/148783836-87db8424-97d3-42fa-a991-282c52e72527.png)
I'd like to close the issue only when ALL linked PRs are successfully merged, not only one of them.

### 2022/02/24 update

Probably, we need to consider cases that PRs are rejected/cancelled and new PRs are added to replace them.  In that case, we could automatically replace the reference prefix to just "refs" to indicate the relevance but not the dependence, along with #4619.

Also we could keep the backward compatibility by introducing new prefixes such as "required by".

**Comment by akylzx** [comment]
Hi [Name or GitHub Team],

Thank you for your continued work on GitHub Issues and Pull Requests — the platform has made major strides in enabling structured workflows for software development.

I'd like to echo and expand on the suggestions made in #4619 regarding more flexible linking between issues and pull requests.

1. Referencing Without Closing
Currently, the only way to semantically link a PR to an issue is via keywords like closes/fixes/resolves, which automatically close the issue upon merge. However, in many workflows, it’s useful to reference an issue from a PR without implying that it should be closed. A dedicated syntax such as refs #1234 or a new keyword like relates to #1234 could improve clarity and intent — especially when the PR is only partially related to the issue.

2. Logical Conditions for Closing Issues
Right now, when multiple PRs reference the same issue using closing keywords, the issue is closed as soon as any one of them is merged. This behaves like a logical OR.

However, in many real-world scenarios, especially in large projects with cross-repo work, a single issue represents a multi-part change that requires all associated PRs to be completed before the issue is resolved. Supporting a logical AND condition — where an issue only closes when all linked PRs with "required by" or similar tags are merged — would significantly improve workflow accuracy and reduce premature closures.

3. Handling Rejected/Cancelled PRs
There’s also the practical consideration of PRs that get closed without being merged. In such cases, it might make sense to automatically downgrade the linkage from "required by" to a non-closing reference (like refs or related to) so the closing condition isn't stuck. Manual override or prompts might also help here.

4. Backward Compatibility and Syntax Suggestions
To preserve existing workflows while expanding flexibility, introducing a new set of keywords could be the way forward:

required by #1234 — PR must be merged to close the issue

refs #1234 — non-closing informational link

relates to #1234 — softer semantic relationship

supersedes #1234 — helpful when replacing prior work

These would be extremely helpful for teams working across multiple codebases, or with complex issue scopes that can't be addressed in a single PR.

Thanks again for your consideration, and for building such a vital platform for collaboration.

Best regards,
